packer/builder/azure/arm/deployment_factory.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.
+// See License.txt in the project root for license information.
+
+package arm
+
+import (
+	"encoding/json"
+
+	"github.com/Azure/azure-sdk-for-go/arm/resources"
+)
+
+type DeploymentFactory struct {
+	template string
+}
+
+func NewDeploymentFactory(template string) DeploymentFactory {
+	return DeploymentFactory{
+		template: template,
+	}
+}
+
+func (f *DeploymentFactory) create(templateParameters TemplateParameters) (*resources.Deployment, error) {
```

Comment:
I am going to make the constructor _internal_ to the class, which hopefully helps with intent when reading.